### PR TITLE
fix: Fix error message for email change confirmation

### PR DESCRIPTION
**Pull request by soniaklimas**
I want to merge this change because it fixes error message for email change confirmation - it returns proper message when token is invalid or email has been already changed and it is used by user.

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

**Review by gajda-w — APPROVED** [review]

File: apps/storefront/src/app/[locale]/(main)/account/profile/_modals/update-email-modal.tsx [hunk 1]
```diff
@@ -9,17 +10,21 @@ import {
 
 import { UpdateEmailForm } from "../_forms/update-email-form";
 
-export async function UpdateEmailModal() {
+export async function UpdateEmailModal({ user }: { user: User | null }) {
```
Comment (GrzegorzDerdak) [inline apps/storefront/src/app/[locale]/(main)/account/profile/_modals/update-email-modal.tsx:13]: Are we using this modal somewhere else than account (where user must be logged in)? If no, then the user will always be here a `User` not `User | null`

**Review by GrzegorzDerdak — APPROVED** [review]

**Review by GrzegorzDerdak — APPROVED** [review]
